### PR TITLE
fix(xcode): update things to make it compatible and buildable with xcode 12.5

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Enhancement**

This PR resolves [CX-]

### Description

With @iskounen, we fixed some issues with Xcode 12.5. That should still work for Xcode 12.4.

## Steps to run:
- run `bundle install` to get the right cocoapods version
- run `bundle e pod install` to get the right pods (maybe add `--repo-update` if you get errors)
- normally run the app (`yarn start` and then play button on xcode)
- don't forget to report back on #practice-mobile if it worked, with your xcode version.

Thank you!

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -537,7 +546,7 @@ PODS:
   - Sentry/Core (6.1.4)
   - Specta (1.0.5)
   - Starscream (3.0.4)
-  - Stripe (14.0.1)
+  - Stripe (14.0.2)
```
Comment (pvinis) [inline Podfile.lock:549]: thats the main change. 14.0.1 had an issue that was not building on code 12.5. first we upgraded this.

File: Artsy.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -4053,11 +4041,6 @@
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 12.1;
-				LIBRARY_SEARCH_PATHS = (
-					"\"$(TOOLCHAIN_DIR)/usr/lib/swift/$(PLATFORM_NAME)\"",
-					"\"$(TOOLCHAIN_DIR)/usr/lib/swift-5.0/$(PLATFORM_NAME)\"",
```
Comment (pvinis) [inline Artsy.xcodeproj/project.pbxproj:4058]: this was the second thing we had to add. this was part of the 0.64 upgrade too, but now we need it even with 0.63 with xcode 12.5. just removing the explicit deps of swift.

File: Gemfile.lock [hunk 3]
```diff
@@ -42,16 +45,15 @@ GEM
       cork
       nap
       open4 (~> 1.3)
-    cocoapods (1.7.5)
-      activesupport (>= 4.0.2, < 5)
+    cocoapods (1.10.1)
```
Comment (pvinis) [inline Gemfile.lock:48]: trying to pod install we got an error for openssl that needs cocoapods 1.9. I unpinned the cocoapods version last week, but I forgot to also update it. another thing that was needed anyway. so now we have 1.10. **remember to run `bundle install` to get the fresh cocoapods version!!**

File: Podfile [hunk 4]
```diff
@@ -162,7 +162,7 @@ end
 # Enables Flipper.
 # Note that if you have use_frameworks! enabled, Flipper will not work and
 # you should disable these next few lines.
-use_flipper!
+use_flipper!({ 'Flipper-Folly' => '2.5.3', 'Flipper' => '0.87.0', 'Flipper-RSocket' => '1.3.1' })
```
Comment (pvinis) [inline Podfile:165]: this is fixing another xcode 12.5 issue that blocks building. Flipper needed updating so we now use a newer version than the one coming with our version of RN.
tip came from https://github.com/facebook/flipper/issues/2215#issuecomment-829053571.
We can remove later.

File: Artsy.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -3408,30 +3408,18 @@
...
 			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/MapboxMobileEvents.framework",
-				"${DWARF_DSYM_FOLDER_PATH}/MapboxMobileEvents.framework.dSYM",
-				"${BUILT_PRODUCTS_DIR}/8F3E82C3-52DF-39E5-809D-3389B6CBCAD4.bcsymbolmap",
-				"${BUILT_PRODUCTS_DIR}/4860631E-761E-36C9-9DA5-4504DE704214.bcsymbolmap",
-				"${BUILT_PRODUCTS_DIR}/0DC8DAD8-C6D7-343B-ACBF-79B48068D7CC.bcsymbolmap",
-				"${BUILT_PRODUCTS_DIR}/85944F06-DDCF-309B-B848-CCA12C2D2A09.bcsymbolmap",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/Mapbox.framework",
-				"${DWARF_DSYM_FOLDER_PATH}/Mapbox.framework.dSYM",
```
Comment (pvinis) [inline Artsy.xcodeproj/project.pbxproj:3432]: these seem to be changes that cocoapods 1.10 does. somehow it connects mapbox differently. less lines is always better, so 🤷.

**Comment by pvinis** [comment]
I guess circleci has still an old xcode version, so if it passes, we should be good.

**Review by iskounen — APPROVED** [review]
These changes work for me locally 👍

**Pull request merged**